frontend/src/libs/components/modal/modal.tsx

```diff
@@ -0,0 +1,62 @@
+import { createRef } from 'react';
+
+import { Icon, Portal } from '~/libs/components/components.js';
+import { useCallback, useEffect } from '~/libs/hooks/hooks.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  children: React.ReactNode;
+  onClose: () => void;
+  isOpen: boolean;
+};
+
+const Modal: React.FC<Properties> = ({
+  isOpen,
+  onClose: handleOnClose,
+  children,
+}) => {
+  const outsideModalDivReference = createRef<HTMLDivElement>();
```

Comment:
Please use `useRef` hook instead, `createRef` is usually used in class components. I guess `overlayRef` would be a better variable name

Reply:
Thanks